java-frontend/src/main/java/org/sonar/java/se/xproc/MethodBehavior.java

```diff
@@ -52,20 +52,27 @@ public MethodBehavior(Symbol.MethodSymbol methodSymbol) {
     this.varArgs = ((JavaSymbol.MethodJavaSymbol) methodSymbol).isVarArgs();
     this.arity = methodSymbol.parameterTypes().size();
   }
-
   public void createYield(ExplodedGraph.Node node) {
+    createYield(node, true);
+  }
+  
+  public void createYield(ExplodedGraph.Node node, boolean storeNodeForReporting) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Provide multiple methods instead of using "storeNodeForReporting" to determine which action to take. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2301)